app/classes/query/base_new.rb

```diff
@@ -0,0 +1,105 @@
+# frozen_string_literal: true
+
+# base class for Query searches
+class Query::BaseNew
+  include ActiveModel::API
+  include ActiveModel::Attributes
+  include ActiveModel::Validations::Callbacks
+
+  include Query::Modules::ClassMethods
+  include Query::ScopeModules::ResultAccessors
+  include Query::ScopeModules::Initialization
+  include Query::ScopeModules::SequenceOperators
+  include Query::Modules::Validation
+
+  validates_with Query::Modules::Validator
+
+  # "clean up" and reassign attributes before validation
+  before_validation :clean_and_validate_params
+
+  attr_writer :record
+
+  def self.parameter_declarations
+    { order_by: :string }
+  end
+
+  delegate :parameter_declarations, to: :class
+
+  # can use has_attribute? here.
+  # only called in ApplicationController::Queries#apply_one_content_filter
+  def self.takes_parameter?(key)
+    parameter_declarations.key?(key)
+  end
+
+  delegate :takes_parameter?, to: :class
+
+  def self.scope_parameters
+    excepts = [:id_in_set, :preference_filter]
+    @scope_parameters = parameter_declarations.except(*excepts).keys +
+                        [:id_in_set]
+  end
+
+  delegate :scope_parameters, to: :class
+
+  def self.content_filter_parameters
+    filters = Query::Filter.all
+    @content_filter_parameters ||= filters.each_with_object({}) do |f, p|
+      p[f.sym] = f.type
+    end.freeze
+  end
+
+  delegate :content_filter_parameters, to: :class
+
+  # Can the current class be called as a subquery of the target Query class?
+  def relatable?(target)
+    self.class.related?(target, model.name.to_sym)
+  end
+
+  def subquery_of(target)
+    self.class.current_or_related_query(target, model.name.to_sym, self)
+  end
+
+  def default_order
+    self.class.default_order ||
+      raise("Didn't supply default order for #{model} query.")
+  end
+
+  def ==(other)
+    serialize == other.try(&:serialize)
+  end
+
+  # NOTE: QueryRecord[:description] is not a Rails-serialized column; we call
+  # `to_json` here to serializate it ourselves.
+  # Prepares the query params, adding the model, for saving to a QueryRecord.
+  # The :description column is accessed not just to recompose a query, but to
+  # identify existing query records that match current params. That's why the
+  # keys are sorted here before being stored as strings in to_json - because
+  # when matching a serialized hash, strings must match exactly. This is
+  # more efficient however than using a Rails-serialized column and comparing
+  # the parsed hashes (in whatever order), because when a column is serialized
+  # you can't use SQL on the column value, you have to compare parsed instances.
+  def serialize
+    # params.sort.to_h.merge(model: model.name).to_json
```

Comment:
Consider moving this line into the prior comment.

Reply:
Ah, good catch. I actually need to delete that. 
That's the version of the same line, from before I added the `ActiveModel::API`.

Reply:
Moved it up, in case needed later